csaf_2.0/prose/csaf-v2-editor-draft.md

```diff
@@ -4598,6 +4598,31 @@ The relevant paths for this test are:
 
 > A tool MAY remove such element as a quick fix.
 
+## 6.2.16 Missing Product Identification Helper
+
+For each element of type `/$defs/full_product_name_t` it must be tested that it includes the property `product_identification_helper` and at least one Product Identification Helper is provided.
```

Comment:
I guess that is the same question as in https://github.com/oasis-tcs/csaf/pull/405#discussion_r801832289 ...

Can we rely on the schema to enforce non-empty objects?

My first intention even was to write:
```suggestion
For each element of type `/$defs/full_product_name_t` it must be tested that at least one Product Identification Helper is provided.
```

Reply:
Another option would be:
```suggestion
For each element of type `/$defs/full_product_name_t` it must be tested that it includes the property `product_identification_helper`.

Reply:
Personally, I tend to option 2...

Reply:
I think changing that strategy of ours now is a bit late down the road. I remember when we discussed the optional elements (most of the CSAF vocabulary). I was in favor for relying on the „do not include null values“ convention as convention, while the TC wanted a more strict approach to ease the receiver side in automation. so, we should trust the schema for cardinality IMO, because it represents important constraints well within the scope of a schema language.